tests/chainer_tests/functions_tests/math_tests/test_clip.py

```diff
@@ -8,6 +8,7 @@
 from chainer import gradient_check
 from chainer import testing
 from chainer.testing import attr
+from chainer.functions.math.clip import (Clip, ClipGrad)
```

Comment:
Actually could you check the input validation using the functions `chainer.functions.clip` and not the `FunctionNode`s? The `FunctionNode`s are sort of "private" although this is not clearly communicated. That's also why I suggested asserting in `ClipGrad`. Another few tips general tips is that we don't import non-modules and we only have single imports per line 🙂